routemaster/state_machine/api.py

```diff
@@ -101,6 +101,37 @@ def create_label(app: App, label: LabelRef, metadata: Metadata) -> Metadata:
     return metadata
 
 
+def restore_label_and_restart(
+    app: App,
+    label: LabelRef,
+    replacement_metadata: Metadata,
+) -> Metadata:
+    """Restores a label that was previously deleted."""
+    state_machine = get_state_machine(app, label)
+
+    row = lock_label(app, label)
+
+    # Record the label as having been restored and add its initial metadata
+    if not row.deleted:
+        raise AssertionError(f"Label {label} is not marked as deleted!")
```

Comment:
This could maybe be `LabelAlreadyExists` or a new exception. Same with the one below as well.

Reply:
5648ff3387d0b02e123b96a9d9ddba07384cd15c